### ISSUE TITLE
Bug: `_middleware.tsx` doesn't run before `_500.tsx` or `_error.tsx`

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/fresh#2573]

**Comment by NanoEdit** [comment]
After #2598 _middleware.ts runs in subfolders but not on the root folder

For example routes folder structure:
- _middleware.ts
- index.tsx
- category/index.tsx

Then at `/category/foo` middleware runs 
at `/foo` it doesn't


BTW _layout is fixed correctly in #2598, but _middleware doesn't

**Comment by csvn** [comment]
Maybe close this issue in favor of #2843? They seem to be duplicates

**Comment by marvinhagemeister** [comment]
Duplicate of https://github.com/denoland/fresh/issues/2843
